tests/foreman/cli/test_bootstrap_script.py

```diff
@@ -15,54 +15,73 @@
-
-
-class BootstrapScriptTestCase(CLITestCase):
-    """Test class for bootstrap script."""
-
-    @classmethod
-    def setUpClass(cls):
-        """create VM for testing """
-        super().setUpClass()
-
-    @pytest.mark.tier1
-    @pytest.mark.stubbed
-    def test_positive_register(self):
-        """System is registered
-
-        :id: e34561fd-e0d6-4587-84eb-f86bd131aab1
-
-        :Steps:
-
-            1. assure system is not registered
-            2. register a system
-
-        :expectedresults: system is registered, host is created
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: Critical
-        """
-
-    @pytest.mark.tier1
-    @pytest.mark.stubbed
-    @pytest.mark.upgrade
-    def test_positive_reregister(self):
-        """Registered system is re-registered
-
-        :id: d8a7aef1-7522-47a8-8478-77e81ca236be
-
-        :Steps:
-
-            1. register a system
-            2. assure system is registered
-            3. register system once again
-
-        :expectedresults: system is newly registered, host is created
-
-        :CaseAutomation: NotAutomated
-
-        :CaseImportance: Critical
-        """
+from broker.broker import VMBroker
+from nailgun import entities
+
+from robottelo.api.utils import promote
+from robottelo.hosts import ContentHost
+from robottelo.test import settings
+
+
+@pytest.fixture(scope='module')
+def activation_key(module_org, module_lce, module_published_cv):
+    promote(module_published_cv.version[0], module_lce.id)
+    module_published_cv = module_published_cv.read()
+    activation_key = entities.ActivationKey(
+        content_view=module_published_cv,
+        environment=module_lce,
+        organization=module_org,
+    ).create()
+    return activation_key
+
+
+@pytest.mark.tier1
+def test_positive_register(
+    module_org, module_location, module_lce, activation_key, module_published_cv
+):
+    """System is registered
+
+    :id: e34561fd-e0d6-4587-84eb-f86bd131aab1
+
+    :Steps:
+
+        1. Ensure system is not registered
+        2. Register a system
+        3. Ensure system is registered
+        4. Register system once again
+
+
+    :expectedresults: system is registered, host is created
+
+    :CaseAutomation: Automated
+
+    :CaseImportance: Critical
+    """
+    hg = entities.HostGroup(location=[module_location], organization=[module_org]).create()
+    with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as vm:
+        # assure system is not registered
+        result = vm.execute('subscription-manager identity')
+        # result will be 1 if not registered
+        assert result.status == 1
+        assert vm.execute(
+            f'curl -o /root/bootstrap.py "http://{settings.server.hostname}/pub/bootstrap.py" '
+        )
+        assert vm.execute(
+            f'python /root/bootstrap.py -s "{settings.server.hostname}" -o {module_org.name} '
+            f'-L Default_Location -a {activation_key.name} --hostgroup={hg.name}'
```

Comment:
@swadeley should we use module_location here?

Reply:
@jameerpathan111 correct, the location here shouldn't be hard-coded, this should be a blocking comment.

Reply:
sorry about that, now fixed